### PR TITLE
[Test] TensorV2 multiplication unit test

### PR DESCRIPTION
**Pull request by djeong20**
This PR includes unit tests for verifying multiplication operations in the TensorV2 class.
Note that all unit tests are the same as tests in unittest_nntrainer_tensor.cpp file.

**Self-evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test:   [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: test/unittest/unittest_nntrainer_tensor_v2_fp16.cpp [hunk 1]
```diff
@@ -394,6 +394,963 @@ TEST(nntrainer_Tensor, initialize_08_p) {
...
+                                 nntrainer::Tdatatype::FP16);
+    nntrainer::TensorV2 t = rangedV2(3, 2, 4, 5, nntrainer::Tformat::NCHW,
+                                     nntrainer::Tdatatype::FP16);
+    nntrainer::TensorV2 m = rangedV2(1, 2, 4, 5, nntrainer::Tformat::NCHW,
+                                     nntrainer::Tdatatype::FP16);
+    _FP16 answer_data[] = {static_cast<_FP16>(0),    static_cast<_FP16>(1),
+                           static_cast<_FP16>(4),    static_cast<_FP16>(9),
+                           static_cast<_FP16>(16),   static_cast<_FP16>(25),
+                           static_cast<_FP16>(36),   static_cast<_FP16>(49),
+                           static_cast<_FP16>(64),   static_cast<_FP16>(81),
```
Comment (djeong20) [inline test/unittest/unittest_nntrainer_tensor_v2_fp16.cpp:486]: all these static_cast to _FP16 is making readability very low.
if anyone has a good idea, please give me a suggestion!

Comment (DonghakPark) [reply test/unittest/unittest_nntrainer_tensor_v2_fp16.cpp:486]: why don't you make Macro or Function that can convert to FP16?

Comment (DonghakPark) [reply test/unittest/unittest_nntrainer_tensor_v2_fp16.cpp:486]: for example 
```cpp
template <typename T> // T models Any
struct static_cast_func
{
  template <typename T1> // T1 models type statically convertible to T
  T operator()(const T1& x) const { return static_cast<T>(x); }
};
```
```cpp
std::transform(char_buff, char_buff + len, float_buff, static_cast_func<float>());
```
like this

**Review by DonghakPark — APPROVED** [review]

File: test/unittest/unittest_nntrainer_tensor_v2.cpp [hunk 2]
```diff
@@ -321,6 +321,433 @@ TEST(nntrainer_Tensor, initialize_08_p) {
...
+
+  status = input.multiply_i(2.0);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  float *data = original.getData<float>();
+  ASSERT_NE(nullptr, data);
+  float *indata = input.getData<float>();
+  ASSERT_NE(nullptr, indata);
+
+  for (int i = 0; i < batch * height * width * channel; ++i) {
```
Comment (jihochu) [inline test/unittest/unittest_nntrainer_tensor_v2.cpp:345]:
```suggestion
  for (int i = 0; i < batch * channel * height * width; ++i) {
```
?

**Review by jihochu — APPROVED** [review]

File: test/unittest/unittest_nntrainer_tensor_v2_fp16.cpp (hunk 1 above)
Comment (djeong20) [reply test/unittest/unittest_nntrainer_tensor_v2_fp16.cpp]: thanks :) this really helped a lot

File: test/unittest/unittest_nntrainer_tensor_v2.cpp (hunk 2 above)
Comment (djeong20) [reply test/unittest/unittest_nntrainer_tensor_v2.cpp]: fixed.

**Review by SeoHyungjun — APPROVED** [review]
LGTM!

**Pull request merged**